pkg/image/airgap.go

```diff
@@ -675,6 +700,114 @@ func reportWriterWithProgress(imageInfos map[string]*imagetypes.ImageInfo, repor
 	return pipeWriter
 }
 
+type OCIArtifactFile struct {
+	Name      string
+	Path      string
+	MediaType string
+}
+
+func PushEmbeddedClusterArtifacts(airgapBundle string, opts imagetypes.PushEmbeddedArtifactsOptions) error {
+	embeddedClusterDir := filepath.Join(airgapBundle, "embedded-cluster")
+	if _, err := os.Stat(embeddedClusterDir); err != nil {
+		if os.IsNotExist(err) {
+			// no embedded-cluster dir, nothing to do
+			return nil
+		}
+		return errors.Wrap(err, "failed to stat embedded-cluster dir")
+	}
+
+	err := filepath.Walk(embeddedClusterDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !info.Mode().IsRegular() {
+			return nil
+		}
+
+		repo := "binary"
+		if info.Name() == "charts.tar.gz" {
+			repo = "charts"
+		} else if info.Name() == "images-amd64.tar" {
+			repo = "images"
+		}
+		repo = filepath.Join("embedded-cluster", repo)
```

Comment:
Would like to get thoughts on this piece. Will the charts and images artifacts have well-known names?

Separately, any thoughts on the repo names that they are pushed to? `embedded-cluster/binary`, `embedded-cluster/charts`, and `embedded-cluster/images`?

Reply:
Iterate over files and push each one as an artifact with a deterministic name